src/test/java/uk/gov/hmcts/reform/ccd/document/am/feign/CaseDocumentClientApiTest.java

```diff
@@ -0,0 +1,240 @@
+import static org.springframework.http.HttpHeaders.CONTENT_TYPE;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static org.springframework.http.MediaType.MULTIPART_FORM_DATA_VALUE;
+
+@SpringBootTest(classes = {CaseDocumentClientApi.class})
+@PropertySource(value = "classpath:application.yml")
+@EnableAutoConfiguration
+@AutoConfigureWireMock(port = 5050)
+class CaseDocumentClientApiTest {
+
+    private static final UUID DOCUMENT_ID = UUID.randomUUID();
+
+    private static final String URL = "/cases/documents";
+
+    private static final String CLASSIFICATION = "aClassification";
+    private static final String CASE_TYPE_ID = "aCaseTypeId";
+    private static final String JURISDICTION = "aJurisdictionId";
+
+    private static final String SERVICE_AUTHORISATION_KEY = "ServiceAuthorization";
+    private static final String BEARER = "Bearer ";
+    private static final String TOKEN = "user1";
+    private static final String SERVICE_AUTHORISATION_VALUE = BEARER + TOKEN;
+    private static final String AUTHORISATION_VALUE = "a Bearer idam token";
+
+    private static final boolean PERMANENT = false;
+    public static final String PERMANENT_QUERY_PARAM = "permanent";
+
+    private MockMultipartFile multipartFile = new MockMultipartFile("testFile1", "content".getBytes());
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Autowired
+    private CaseDocumentClientApi caseDocumentClientApi;
+
+    @BeforeEach
+    void setUp() {
+        objectMapper.registerModule(new JavaTimeModule());
+        objectMapper.registerModule(new ParameterNamesModule());
+        objectMapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+    }
+
+    @Test
+    void shouldSuccessfullyUploadDocuments() {
+        List<MultipartFile> files = new ArrayList<>();
+        files.add(multipartFile);
+
+        DocumentUploadRequest request = new DocumentUploadRequest(CLASSIFICATION,
+                                                                  CASE_TYPE_ID, JURISDICTION, files);
+        stubForUpload(request);
+
+        ResponseEntity responseEntity = caseDocumentClientApi.uploadDocuments(
+            AUTHORISATION_VALUE, SERVICE_AUTHORISATION_VALUE, request);
+
+        assertEquals(HttpStatus.OK, responseEntity.getStatusCode());
+    }
+
+    @Test
+    void shouldSuccessfullyGetDocumentBinary() throws IOException {
+        ResponseEntity response = new ResponseEntity(HttpStatus.OK);
+
+        stubForDocumentBinary(response);
+
+        ResponseEntity finalResponse = caseDocumentClientApi.getDocumentBinary(
+            TOKEN,
+            SERVICE_AUTHORISATION_VALUE,
+            DOCUMENT_ID
+        );
+
+        assertEquals(HttpStatus.OK, finalResponse.getStatusCode());
+    }
+
+    @Test
+    void shouldSuccessfullyGetMetadataForDocument() throws IOException {
+        Date createdOn = Date.from(Instant.now());
+
+        Document response = new Document();
```

Comment:
please set all other fields and assert them. Define a builder in Document if it's not there..